### PR TITLE
Implements #19825: SqlConnection doesn't override DbProviderFactory property

### PR DESCRIPTION
**Pull request by FransBouma**
This PR implements #19825 with an override of the SqlConnection.DbProviderFactory property by returning the static SqlClientFactory.Instance object. 

cc @karelz @saurabh500

### CODE REVIEW

File: src/System.Data.SqlClient/src/System/Data/SqlClient/SqlConnection.cs [hunk 1]
```diff
@@ -317,6 +317,11 @@ public string WorkstationId
             }
         }
 
+        protected override DbProviderFactory DbProviderFactory
+        {
+            get { return SqlClientFactory.Instance; }
+        }
```
Comment (stephentoub) [inline src/System.Data.SqlClient/src/System/Data/SqlClient/SqlConnection.cs]: Can you add tests for this?

Comment (FransBouma) [inline src/System.Data.SqlClient/src/System/Data/SqlClient/SqlConnection.cs]: Sure, will do that tomorrow :)

**Comment by karelz** [comment]
Thanks a lot @FransBouma! I sent you Collaborator invite. Once you accept, we will be able to assign things to you even if you didn't create them.
BTW: It will auto-subscribe you to the entire repo - just disabled it on GH, as it feels like DoS attack on your inbox ;) (see recent tweets from Nick Craver & Ben Adams)

**Comment by FransBouma** [comment]
@karelz  haha yeah 'DoS' is a nice description :) Have unwatched it for now, flood is a bit massive ;)

**Comment by saurabh500** [comment]
Thanks @FransBouma 
For tests, I dont think a test for this change would require server interactions. Feel free to add test(s) to `FunctionalTests` of SqlClient.

**Comment by FransBouma** [comment]
@saurabh500 yeah a simple reflection based read of the property will suffice I think. Learned a lot with this small PR to get started with the process :)

**Comment by FransBouma** [comment]
I've run into a problem: I have written the test, but can't run it from within visual studio: the xunit front controller crashes when running the test with latest R# test runner, as it can't find xunit.dll or xunit.execution.*.dll and visual studio 2017 (15.2) doesn't find any tests in the solution, no matter what I try. Googled to find a solution for this but came up with nothing. 

I want to run this just test and not the 1000s of other tests of course.

**Comment by FransBouma** [comment]
I've committed the test, after I managed to run it (and pass it) using the debugger, by setting the functional tests project as the startup project, a breakpoint in the test and then manually continuing through all the exceptions that were thrown because I didn't setup a connection to a server. Of course not the experience one wants to have, but until this test runner issue is resolved it's all I had. 

Hopefully it's fixable. No idea if I can configure xunit (I always use nunit, first time with xunit) to run just 1 test, not ideal as well, but better than wading through a sea of failing tests before arriving to your own test ;) 

Do I need to install the 15.3 preview perhaps? Or is it an issue with xunit and its runner?

**Comment by karelz** [comment]
@mellinoe @weshaggard can you help get @FransBouma unblocked running tests?

**Comment by FransBouma** [comment]
@mellinoe @weshaggard @karelz it's a known issue I found out, issues aren't showing up in vs2017 yet due to updated coreclr. I didn't know that, hence the struggle. I have now learned how to do that properly, also how to setup vs2017 to debug just 1 test. Will add that to the wiki page now for contributors so they get started right away with the right settings :)

**Comment by karelz** [comment]
Sorry for the trouble. I guess using VS for CoreFX/CoreCLR development is not the happy path yet :(
Thanks for updating the wiki -- we can ask on [new-contrib docs gitter](https://gitter.im/dotnet/corefx-contrib-docs?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge) to try & document VS path as well ...

**Comment by saurabh500** [comment]
@FransBouma I think this makes sense with the DbProviderFactories work that you are planning and doesn't make sense as a standalone change in 2.0. 
So we can merge this in master and this doesn't need to be ported over to `release/2.0.0` branch. 
Let me know if you think otherwise.

**Comment by FransBouma** [comment]
The change is indeed solely needed for DbProviderFactories, the property that's overridden isn't used otherwise, AFAIK

**Comment by saurabh500** [comment]
@dotnet-bot test this please

File: src/System.Data.SqlClient/tests/FunctionalTests/SqlConnectionTest.cs [hunk 2]
```diff
@@ -22,6 +24,8 @@ public void ConnectionTest()
             }
         }
 
+
+        [Fact]
```
Comment (saurabh500) [inline src/System.Data.SqlClient/tests/FunctionalTests/SqlConnectionTest.cs]: @FransBouma I missed deleting the `[Fact]` attribute while resolving merge conflicts. 
Can you push a change by deleting `[Fact]` line from this test ?

Comment (saurabh500) [inline src/System.Data.SqlClient/tests/FunctionalTests/SqlConnectionTest.cs]: I realized that I can make the change myself :) Sorry about the confusion

**Comment by saurabh500** [comment]
@dotnet-bot test Innerloop OSX10.12 Debug x64 Build and Test please.
Test Innerloop OSX10.12 Release x64 Build and Test please
There was an I/O error during the build. This could be caused by a hung process on the target machine, preventing it from cleaning up the workspace, or other issues, such as low disk space.

**Comment by saurabh500** [comment]
@dotnet-bot Test Innerloop Ubuntu14.04 Debug x64 Build and Test  please

The build timed out, or was forcibly aborted after reaching the maximum length. View the log right before the flagged timeout message to see what was happening. Build timed out (after 120 minutes). Marking the build as aborted.

**Pull request merged**